pkg/controller/s3/bucket/replicationConfig_test.go

```diff
@@ -459,3 +460,453 @@ func TestReplicationLateInit(t *testing.T) {
+							AccessControlTranslation: &v1beta1.AccessControlTranslation{Owner: owner},
+							Account:                  &accountID,
+							Bucket:                   &bucketName,
+							EncryptionConfiguration:  &v1beta1.EncryptionConfiguration{ReplicaKmsKeyID: &kmsID},
+							Metrics: &v1beta1.Metrics{
+								EventThreshold: v1beta1.ReplicationTimeValue{Minutes: int32(replicationTime)},
+								Status:         enabled,
+							},
+							ReplicationTime: &v1beta1.ReplicationTime{
+								Status: enabled,
+								Time:   v1beta1.ReplicationTimeValue{Minutes: int32(replicationTime)},
+							},
+							StorageClass: &storage,
+						},
+						ExistingObjectReplication: &v1beta1.ExistingObjectReplication{Status: enabled},
+						Filter: &v1beta1.ReplicationRuleFilter{
+							And: &v1beta1.ReplicationRuleAndOperator{
+								Prefix: &prefix,
+								Tags:   tags,
+							},
+						},
+						ID:                      &id,
+						Priority:                priority,
+						SourceSelectionCriteria: &v1beta1.SourceSelectionCriteria{SseKmsEncryptedObjects: v1beta1.SseKmsEncryptedObjects{Status: enabled}},
+						Status:                  "Disabled",
+					}},
+				},
+				b: &s3types.ReplicationConfiguration{
+					Role: &role,
+					Rules: []s3types.ReplicationRule{{
+						DeleteMarkerReplication: &s3types.DeleteMarkerReplication{Status: s3types.DeleteMarkerReplicationStatusEnabled},
+						Destination: &s3types.Destination{
+							AccessControlTranslation: &s3types.AccessControlTranslation{Owner: s3types.OwnerOverrideDestination},
+							Account:                  &accountID,
+							Bucket:                   &bucketName,
+							EncryptionConfiguration:  &s3types.EncryptionConfiguration{ReplicaKmsKeyID: &kmsID},
+							Metrics: &s3types.Metrics{
+								EventThreshold: &s3types.ReplicationTimeValue{Minutes: int32(replicationTime)},
+								Status:         s3types.MetricsStatusEnabled,
+							},
+							ReplicationTime: &s3types.ReplicationTime{
+								Time:   &s3types.ReplicationTimeValue{Minutes: int32(replicationTime)},
+								Status: s3types.ReplicationTimeStatusEnabled,
+							},
+							StorageClass: s3types.StorageClassOnezoneIa,
+						},
+						ExistingObjectReplication: &s3types.ExistingObjectReplication{Status: s3types.ExistingObjectReplicationStatusEnabled},
+						Filter: &s3types.ReplicationRuleFilterMemberAnd{
+							Value: s3types.ReplicationRuleAndOperator{
+								Prefix: &prefix,
+								Tags:   awsTags,
+							},
+						},
+						ID:                      &id,
+						Priority:                priority,
+						SourceSelectionCriteria: &s3types.SourceSelectionCriteria{SseKmsEncryptedObjects: &s3types.SseKmsEncryptedObjects{Status: s3types.SseKmsEncryptedObjectsStatusEnabled}},
+						Status:                  s3types.ReplicationRuleStatusEnabled,
+					}},
+				},
+			},
+			want: want{
+				isUpToDate: 1,
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			g := NewGomegaWithT(t)
```

Comment:
Would you mind updating this to remove gomega? Should be a small diff - same approach as https://github.com/crossplane/provider-aws/pull/963/commits/aaaa66b9725b523d2bc779119ec3885ce030de2e